### DISCUSSION TITLE
Could you please make it easier to test without `trio`?

### DISCUSSION THREAD

**Discussion by mgorny** [encode/httpx#3201] [Ideas]
I'm working on adding Python 3.13 support to `httpx` in Gentoo. Unfortunately, `trio` right now fails tests with Python 3.13 pretty badly, so I'd like to skip it for the time being. Also, we don't really need it, except to run the tests. I can easily work through most of the issues but the direct `import trio` in `tests/concurrency.py` is quite a hassle. Would you consider accepting a patch that would move it into the function, e.g.:

```diff
diff --git a/tests/concurrency.py b/tests/concurrency.py
index a8ed558..d5ce803 100644
--- a/tests/concurrency.py
+++ b/tests/concurrency.py
@@ -5,11 +5,11 @@ Async environment-agnostic concurrency utilities that are only used in tests.
 import asyncio
 
 import sniffio
-import trio
 
 
 async def sleep(seconds: float) -> None:
     if sniffio.current_async_library() == "trio":
+        import trio
         await trio.sleep(seconds)  # pragma: no cover
     else:
         await asyncio.sleep(seconds)
```

so that we could simply deselect tests using `trio`, without getting a top-level failure about the missing package?

**Comment by zanieb** [comment]
Seems reasonable to me. Is this the only place that would need to change?

**Reply by mgorny** [reply]
Yes. I would also need to do `pytest -o filterwarnings= -k "not trio"` but that's not really a problem for us.
